### ISSUE TITLE
Contributor PR fails CircleCI job `verify-accessibility-results`

### ISSUE THREAD

**Issue by MikeMcC399** [cypress-io/cypress#30471]
### Current behavior

PRs from forks fail [.circleci/workflows.yml](https://github.com/cypress-io/cypress/blob/develop/.circleci/workflows.yml) job `verify-accessibility-results`.

It also looks like the job is being started before the project has been built.

![image](https://github.com/user-attachments/assets/30fb3de0-7dc7-4a06-b233-3c13ac864984)

### Desired behavior

According to the workflow code it is intended for contributor PRs to skip the `verify-accessibility-results` job, however there seems to be an issue with the logic:

https://github.com/cypress-io/cypress/blob/d5dfac5882d2495475747d402765dbcfbb119efd/.circleci/workflows.yml#L1467-L1472

### Test code to reproduce

Submit PR from fork.

- https://github.com/cypress-io/cypress/pull/30472

### Cypress Version

`13.15.1` `develop` branch `HEAD` at https://github.com/cypress-io/cypress/commit/d5dfac5882d2495475747d402765dbcfbb119efd

### Node version

v18.17.1

### Operating System

Ubuntu 20.04.6 LTS

### Debug Logs

- See https://app.circleci.com/pipelines/github/cypress-io/cypress/65004/workflows/09783049-8213-4eae-bac8-015f06d2c8dc/jobs/2702806 from PR https://github.com/cypress-io/cypress/pull/30472

### Other

- The job `verify-accessibility-results` was added by PR https://github.com/cypress-io/cypress/pull/30325
